federation/pkg/federation-controller/util/federated_informer_test.go

```diff
@@ -92,7 +92,18 @@ func TestFederatedInformer(t *testing.T) {
 			framework.ResourceEventHandlerFuncs{})
 	}
 
-	informer := NewFederatedInformer(fakeClient, targetInformerFactory).(*federatedInformerImpl)
+	addedClusters := make(chan string, 5)
+	deletedClusters := make(chan string, 5)
+	lifecycle := ClusterLifecycleHandlerFuncs{
+		ClusterAvailable: func(cluster *federation_api.Cluster, _ FederationView, _ FederatedReadOnlyStore) {
+			addedClusters <- cluster.Name
```

Comment:
I would add:
close(addedClusters)
after it, to test that nothing else is delivered after.